### PR TITLE
Add ExistentialToGeneric mangling and demangling code

### PR DESCRIPTION
**Pull request by rajbarik**
Add a new mangler/demangler option 'e' for ExistentialToGeneric needed for 
https://github.com/apple/swift/pull/13991

### CODE REVIEW

**Review by eeckstein — CHANGES_REQUESTED** [review]
Can you please also update docs/ABI/Mangling.rst

File: lib/Demangling/Remangler.cpp [hunk 1]
```diff
@@ -1019,7 +1019,20 @@ void Remangler::mangleFunctionSignatureSpecializationParam(Node *node) {
       Buffer << 'x';
       return;
     default:
-      if (kindValue & unsigned(FunctionSigSpecializationParamKind::Dead)) {
+      if (kindValue &
+          unsigned(
+              FunctionSigSpecializationParamKind::ExistentialToGeneric)) {
+        Buffer << 'e';
```
Comment (slavapestov) [inline lib/Demangling/Remangler.cpp]: It seems weird that this is an if/else if construct that duplicates some of the cases here. Can you instead redo this as a series of independent ifs instead?

**Comment by rajbarik** [comment]
@eeckstein please see the updated doc

**Review by eeckstein — APPROVED** [review]

File: lib/Demangling/Remangler.cpp (hunk 1 above)
Comment (slavapestov) [reply lib/Demangling/Remangler.cpp]: Ok. It seems strange that the first letter is lower case and the rest are upper case, for each of the different argument kinds. But I guess that’s what we have today already.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**